lib/widgets/cards/notification_card.dart

```diff
@@ -0,0 +1,191 @@
+import 'dart:developer';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+
+import 'package:fokus/model/currency_type.dart';
+import 'package:fokus/services/app_locales.dart';
+import 'package:fokus/utils/app_paths.dart';
+import 'package:fokus/utils/icon_sets.dart';
+import 'package:fokus/utils/theme_config.dart';
+import 'package:fokus/widgets/chips/attribute_chip.dart';
+import 'package:fokus/widgets/cards/item_card.dart';
+
+import 'package:intl/intl.dart';
+
+enum NotificationType{
+  caregiver_receivedAward,
+  caregiver_finishedTaskUngraded,
+  caregiver_finishedTaskGraded,
+  caregiver_unfinishedPlan,
+  child_taskGraded,
+  child_receivedBadge
+}
+
+extension NotificationTypeExtension on NotificationType {
+  static const String _pageKey = 'page.notifications.content';
+
+  String title(BuildContext context, String childName) => {
+    NotificationType.caregiver_receivedAward: childName + " " + AppLocales.of(context).translate("$_pageKey.caregiver.receivedAward"),
+    NotificationType.caregiver_finishedTaskUngraded: childName + " " + AppLocales.of(context).translate("$_pageKey.caregiver.finishedTask"),
+    NotificationType.caregiver_finishedTaskGraded: childName + " " + AppLocales.of(context).translate("$_pageKey.caregiver.finishedTask"),
+    NotificationType.caregiver_unfinishedPlan: childName + " " + AppLocales.of(context).translate("$_pageKey.caregiver.unfinishedPlan"),
+    NotificationType.child_taskGraded: AppLocales.of(context).translate("$_pageKey.child.taskGraded"),
+    NotificationType.child_receivedBadge: AppLocales.of(context).translate("$_pageKey.child.receivedBadge")
```

Comment:
Maybe let's do it the usual way: argument in i18n json and key here. You pass the argument for all the types and it will get ingored when not needed.